src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
Better do `WeightVec{S<:Real, T<:Real, V<:AbstractVector{T}} <: RealVector{T}`. That should simplify constructors since the parameter could be inferred from the input arguments.